src/int/encoding.rs

```diff
@@ -0,0 +1,51 @@
+//! Const-friendly decoding/encoding operations for [`Int`].
+
+use crate::{Int, Uint};
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    /// Create a new [`Int`] from the provided big endian bytes.
+    ///
+    /// See [`Uint::from_be_slice`] for more details.
+    pub const fn from_be_slice(bytes: &[u8]) -> Self {
+        Self(Uint::from_be_slice(bytes))
+    }
+
+    /// Create a new [`Int`] from the provided big endian hex string.
+    ///
+    /// Panics if the hex is malformed or not zero-padded accordingly for the size.
+    ///
+    /// See [`Uint::from_be_hex`] for more details.
+    pub const fn from_be_hex(hex: &str) -> Self {
+        Self(Uint::from_be_hex(hex))
+    }
```

Comment:
I think this should work for adding two's complement encoding support?

```suggestion
    pub const fn from_be_hex(hex: &str) -> Self {
        Self::from_bits(Uint::from_be_hex(hex))
    }
```

Reply:
This wouldn't change things:

```rust
/// Const-friendly [`Int`] constructor.
///
/// Reinterprets the bits of a value of type [`Uint`] as an [`Int`].
/// For a proper conversion, see [`Int::new_from_abs_sign`];
/// the public interface of this function is available at [`Uint::as_int`].
pub(crate) const fn from_bits(value: Uint<LIMBS>) -> Self {
    Self(value)
}
```

Reply:
Aah, right